ui/apps/dashboard/e2e/config/test-utils.ts

```diff
@@ -0,0 +1,384 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Page } from '@playwright/test';
+import * as k8s from '@kubernetes/client-node';
+import { V1ObjectMeta } from '@kubernetes/client-node';
+import { parse } from 'yaml';
+import { expect } from '@playwright/test';
+import {
+    createKarmadaApiClient,
+    setMonacoEditorContent,
+    debugScreenshot,
+    waitForResourceInList,
+    setupDashboardAuthentication,
+    getResourceNameFromYaml
+} from '../test-utils';
+import type { DeepRequired } from '../test-utils';
+
+// Re-export ALL utilities from parent test-utils for single-point import
+export {
+    setupDashboardAuthentication,
+    setMonacoEditorContent,
+    debugScreenshot,
+    waitForResourceInList,
+    getResourceNameFromYaml,
+    createKarmadaApiClient
+};
+
+// Re-export types separately
+export type { DeepRequired };
+
+// ConfigMap & Secret Resource Configurations
+export const RESOURCE_CONFIGS = {
+    configmap: {
+        kind: 'ConfigMap',
+        apiClientConstructor: k8s.CoreV1Api,
+        yamlType: k8s.V1ConfigMap,
+        createMethod: 'createNamespacedConfigMap' as const,
+        deleteMethod: 'deleteNamespacedConfigMap' as const,
+    },
+    secret: {
+        kind: 'Secret',
+        apiClientConstructor: k8s.CoreV1Api,
+        yamlType: k8s.V1Secret,
+        createMethod: 'createNamespacedSecret' as const,
+        deleteMethod: 'deleteNamespacedSecret' as const,
+    }
+} as const;
+
+/**
+ * Generic function to create K8s configmap & secret resources
+ */
+export async function createK8sResource<ResourceType extends keyof typeof RESOURCE_CONFIGS>(
+    resourceType: ResourceType,
+    yamlContent: string
+): Promise<void> {
+    const config = RESOURCE_CONFIGS[resourceType];
+    try {
+        const yamlObject = parse(yamlContent) as {
+            'kind'?: string;
+            'metadata'?: V1ObjectMeta;
+        };
+
+        // Ensure namespace is always defined
+        const namespace = yamlObject.metadata?.namespace || 'default';
+
+        // Ensure metadata object exists
+        if (!yamlObject.metadata) {
+            yamlObject.metadata = {};
+        }
+        yamlObject.metadata.namespace = namespace;
+
+        const k8sApi = createKarmadaApiClient(config.apiClientConstructor);
+
+        // Use the configured createMethod to create the resource
+        await (k8sApi[config.createMethod] as any)({
+            namespace: namespace,
+            body: yamlObject
+        });
```

Comment:
rm `as any`, it seems that there is no need to cast type as any,  change it like following:
```
await (k8sApi[config.createMethod])({
      namespace: namespace,
      body: yamlObject
  });
```